### PR TITLE
Add sass files

### PR DESCRIPTION
**Pull request by maya**
This adds basic sass file structure and sass files into these folders: core, fundamentals, and patterns.

- **core**: base, defaults, variables, grid
- **elements** (basic HTML elements): typography, form, list, table
- **components**: header, navigation, footer, search, alerts 

Components were left blank and added for file structure.

### CODE REVIEW

File: assets/_scss/core/_defaults.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+$h3-font-size:      rem(24px) !default;
+$h4-font-size:      rem(18px) !default;
+$h5-font-size:      rem(16px) !default;
+$h6-font-size:      rem(14px) !default;
+$base-line-height:    1.5em !default;
+$heading-line-height: 1.3em !default;
+
+$font-sans:       'Source Sans Pro', 'Helvetica', 'Arial', sans-serif !default;
+$font-serif:      'Georgia', 'Times New Roman', serif;
+$font-mono:       'Source Code Pro', 'Courier New', 'Courier', monospace !default;
```
Comment (juliaelman) [inline assets/_scss/core/_defaults.scss]: Are we still using `Source Code Pro` or should this be removed?

File: assets/_scss/core/_defaults.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+$tablet-up:       600px !default;
+$widescreen-up:     1201px !default;
+
+
+// Magic Numbers
+$site-max-width:    960px !default;
+$site-margins:      40px !default;
+$article-max-width:   600px !default;
+$border-radius:     0px !default;
+$box-shadow:      0 0 2px rgba(#000, .3) !default;
```
Comment (juliaelman) [inline assets/_scss/core/_defaults.scss]: Nitpicky fix. Please align the values on rows 42 - 52 (similarly to how you did the color values.

File: _config.yml [hunk 3]
```diff
@@ -28,5 +28,3 @@ google_analytics_ua: UA-48605964-19
 
 sass:
   sass_dir: assets/_scss
-  style: :compressed
```
Comment (juliaelman) [inline _config.yml]: Please put the compression back in for production.

File: assets/_scss/core/_variables.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+$h2-font-size:      rem(28px);
+$h3-font-size:      rem(24px);
+$h4-font-size:      rem(18px);
+$h5-font-size:      rem(16px);
+$h6-font-size:      rem(14px);
+$base-line-height:    1.5em;
+$heading-line-height: 1.3em;
+
+$font-sans:       'Source Sans Pro', 'Helvetica', 'Arial', sans-serif;
+$font-mono:       'Source Code Pro', 'Courier New', 'Courier', monospace;
```
Comment (juliaelman) [inline assets/_scss/core/_variables.scss]: Are we still using `Source Code Pro` or should this be removed?

File: assets/_scss/core/_variables.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+$tablet-up:       600px;
+$widescreen-up:     1201px;
+
+
+// Magic Numbers
+$site-max-width:    960px;
+$site-margins:      40px;
+$article-max-width:   600px;
+$border-radius:     0px;
+$box-shadow:      0 0 2px rgba(#000, .3);
```
Comment (juliaelman) [inline assets/_scss/core/_variables.scss]: Nitpicky fix. Please align the values on rows 36 - 46 (similarly to how you did the color values.

**Comment by juliaelman** [comment]
Great work @maya! I made some comments in the code base and also have a few other points:

1. It might be good for us change the filename of `assets/core/_base.scss` to something else, as it might cause confusion with the top level `base.scss` file. 
2. We'll need to do some additional configuration to change our syntax highlighting from `rouge` to `prism.js`. Assuming that we'd like syntax highlighting in markdown files, I think we'll need to add the following plugin to make it work [https://github.com/gmurphey/jekyll-prism-plugin](https://github.com/gmurphey/jekyll-prism-plugin).

File: _config.yml (hunk 3 above)
Comment (maya) [inline _config.yml]: @juliaelman Sure! Can you show what you were thinking about adding in a variable for production and development here? The only thing I could imagine is a separate config for dev and production, but you mentioned there's another way.

File: assets/_scss/core/_defaults.scss (hunk 2 above)
Comment (maya) [inline assets/_scss/core/_defaults.scss]: Fixed in 222e2e7.

File: assets/_scss/core/_variables.scss (hunk 5 above)
Comment (maya) [inline assets/_scss/core/_variables.scss]: Fixed in b1830dd.

File: assets/_scss/core/_defaults.scss (hunk 1 above)
Comment (maya) [inline assets/_scss/core/_defaults.scss]: No we're not. Prism.css includes a font-stack for code: `font-family: Consolas, Monaco, 'Andale Mono', monospace;`. Should we just remove the `$font-mono` variable?

**Comment by maya** [comment]
Thanks for the review @juliaelman! 

Answering your first point:
I noticed Bourbon Refills uses an `all.scss` file to import their scss. What do you think about changing `assets/_scss/_base.scss` to this? Then we can keep `assets/_scss/core/_base.scss` as is.

File: assets/_scss/core/_defaults.scss (hunk 1 above)
Comment (juliaelman) [inline assets/_scss/core/_defaults.scss]: Unless you are planning on using that variable elsewhere, I'd say removing that variable sounds like a good solution.

File: _config.yml (hunk 3 above)
Comment (juliaelman) [inline _config.yml]: I added an issue here: https://github.com/18F/govt-wide-patternlibrary/issues/28 let's go ahead and put it back in for now and address this in that issue. :+1:

**Comment by juliaelman** [comment]
@maya what do you think about changing `assets/_scss/core/_base.scss` to `assets/_scss/core/_core.scss`?

**Comment by maya** [comment]
@juliaelman to me it's more confusing to have a file name repeated inside the same folder. Calling it "all" is semantic bc it demonstrates that all of the files are being imported.

File: assets/_scss/core/_defaults.scss (hunk 1 above)
Comment (maya) [inline assets/_scss/core/_defaults.scss]: Fixed and removed in cbd823e.

File: _config.yml (hunk 3 above)
Comment (maya) [inline _config.yml]: Done in 8e55dbe.

**Comment by juliaelman** [comment]
@maya that makes sense :+1: `all.scss` it is!

**Comment by maya** [comment]
@juliaelman Great! Done in a2ed301.

**Comment by maya** [comment]
@juliaelman Can we chunk out the prism.js task into another issue? I think adding in a jekyll plugin is a large enough task to separate out from this PR.

**Comment by juliaelman** [comment]
@maya sure, but I would then ask to remove `prism.js` from this pull request and open a separate one specifically for changing our code highlighting.

**Comment by maya** [comment]
@juliaelman Good call. Done in ae63159. New issue opened here: https://github.com/18F/govt-wide-patternlibrary/issues/31

**Comment by juliaelman** [comment]
:rocket: looks great and ready to merge!

**Pull request merged**